### PR TITLE
Added Doctrine adapter

### PR DESCRIPTION
**Pull request by pierredup**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,8 @@
     },
     "suggest": {
         "illuminate/contracts": "Required to use the illuminate adapter.",
-        "illuminate/support": "Enables Laravel support."
+        "illuminate/support": "Enables Laravel support.",
+        "doctrine/cache": "Required to use the doctrine adapter."
```
Comment (GrahamCampbell) [inline composer.json]: This needs to be alphabetically sorted. I can fix that after merge though.

**Comment by GrahamCampbell** [comment]
@pierredup Can I ask what you're use case for this is. I'm interested. :)

**Comment by pierredup** [comment]
@GrahamCampbell I have a CLI tool that we use for various tasks at work (e.g deployments), which already uses Doctrine cache. I want to add an `analyze` command that we can use as part of our build process (we have an internal repository and build server, so I can't use the hosted version of StyleCI) and it would be easier to use our existing doctrine cache adapter than having to add more dependencies (Illuminate) for this.

I have already added this to our project that we use, but thought it might be useful for other people as well

**Comment by GrahamCampbell** [comment]
Sure. Thanks for the contribution.

**Pull request merged**